src/main/kotlin/no/nav/klage/oppgave/clients/OppgaveClient.kt

```diff
@@ -1,2 +1,119 @@
+import no.nav.klage.oppgave.domain.Oppgave
+import no.nav.klage.oppgave.service.HjemmelParsingService
+import no.nav.klage.oppgave.utils.getLogger
+import no.nav.klage.oppgave.utils.getSecureLogger
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.http.MediaType
+import org.springframework.retry.annotation.Retryable
+import org.springframework.stereotype.Component
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.WebClientResponseException
+import org.springframework.web.reactive.function.client.bodyToMono
+
+@Component
+class OppgaveClient(
+        private val oppgaveWebClient: WebClient,
+        private val tracer: Tracer,
+        private val hjemmelParsingService: HjemmelParsingService,
+        @Value("\${spring.application.name}") private val applicationName: String
+) {
+
+    companion object {
+        @Suppress("JAVA_CLASS_ON_COMPANION")
+        private val logger = getLogger(javaClass.enclosingClass)
+        private val securelogger = getSecureLogger()
+
+        const val STATUSKATEGORI_AAPEN = "AAPEN"
+        const val HJEMMEL = "HJEMMEL"
+    }
+
+    @Retryable
+    fun hjemmelMagic() {
+        val limit = 10
+        var offset = 0
+
+        val oppgaveResponse = logTimingAndWebClientResponseException("getOppgaver ($offset)") {
+            oppgaveWebClient.get()
+                    .uri { uriBuilder ->
+                        uriBuilder.queryParam("limit", limit)
+                        uriBuilder.queryParam("offset", offset)
+                        uriBuilder.build()
+                    }
+                    .header("X-Correlation-ID", tracer.currentSpan().context().traceIdString())
+                    .header("Nav-Consumer-Id", applicationName)
+                    .retrieve()
+                    .bodyToMono<OppgaveResponse>()
+                    .block() ?: throw RuntimeException("Oppgaver could not be fetched")
+        }
+
+        val oppgaverWithNewHjemmel = setHjemmel(oppgaveResponse.oppgaver)
+
+        oppgaverWithNewHjemmel.forEach { putOppgave(it) }
+
+        offset += limit
+    }
+
+    @Retryable
+    fun putOppgave(oppgave: Oppgave) {
+        logTimingAndWebClientResponseException("putOppgave") {
+            oppgaveWebClient.put()
+                    .uri { uriBuilder ->
+                        uriBuilder.pathSegment("{id}").build(oppgave.id)
+                    }
+                    .contentType(MediaType.APPLICATION_JSON)
+                    .header("X-Correlation-ID", tracer.currentSpan().context().traceIdString())
+                    .header("Nav-Consumer-Id", applicationName)
+                    .bodyValue(oppgave)
+                    .retrieve()
+                    .bodyToMono<Oppgave>()
+                    .block() ?: throw java.lang.RuntimeException("Oppgave could not be put")
+        }
+    }
+
+    private fun setHjemmel(oppgaver: List<Oppgave>): List<Oppgave> =
```

Comment:
Jeg tror jeg ville gjort dette utenfor webklienten. Å kalle på en service med business-logikk herfra føles som at man går bort fra lagdeling der man kun kaller nedover i lagene. 

Tror jeg ville gått for en `OppgaveService` e.l. som kallet nedover for å hente oppgaver, utførte business-logikk, og så kalte ned for lagring igjen. Det er et spørmål om vi trenger en `OppgaveRepository` som jeg later som at jeg har nå, eller om det holder med å rett mot webklienten. Men lagdelingen blir vel det samme.

I min kode så bruker jeg `HjemmelParsingService` for å hente ut hjemmel, og så kaller jeg repo for å lagre.